python/cudf/cudf/core/dtypes.py

```diff
@@ -211,3 +212,32 @@ def __eq__(self, other):
 
     def __repr__(self):
         return f"StructDtype({self.fields})"
+
+
+class DecimalDtype(ExtensionDtype):
+
+    name = "decimal"
+    _metadata = ("precision", "scale")
+
+    def __init__(self, precision, scale):
+        self._typ = pa.decimal128(precision, scale)
+
+    @property
+    def precision(self):
+        return self._typ.precision
+
+    @property
+    def scale(self):
+        return self._typ.scale
+
+    @property
+    def type(self):
+        # might need to account for precision and scale here
+        return decimal.Decimal
```

Comment:
I believe the python `decimal` library doesn't allow setting the precision / scale when constructing objects as it does both fixed point and floating point decimal math. Maybe we should consider an alternative object here?

Reply:
Yup -- open to suggestions here. `.type` seems to be a question mark for all our extension dtypes...

Reply:
Agreed. We can leave this for now I suppose, but we really should have some type of class here that allows us to construct `cudf.Scalar` objects in the future.

Reply:
Just `cudf.Scalar`, I suppose?

Reply:
The fear in doing that is I believe there's a lot of code out there that uses `.type` expecting distinct classes for different Scalars. I.E. this is in PyArrow: https://github.com/apache/arrow/blob/f40507545c9c303e85799a874833a4ab48556071/python/pyarrow/pandas_compat.py#L83-L103

Reply:
Makes sense. Punt for now?

Reply:
Yea I suppose. I think the answer is we need classes that wrap our new Scalar class that are strongly typed.

Reply:
Yeah could be like `DecimalDtypeType` for example.